core/src/main/java/org/modelmapper/internal/ProxyFactory.java

```diff
@@ -62,15 +64,57 @@
       throw errors.invocationAgainstFinalClass(type).toException();
 
     try {
-      return OBJENESIS.newInstance(new ByteBuddy()
+      final DynamicType.Unloaded<T> unloaded = new ByteBuddy()
           .subclass(type)
           .method(METHOD_FILTER)
           .intercept(InvocationHandlerAdapter.of(interceptor))
-          .make()
-          .load(useOSGiClassLoaderBridging ? BridgeClassLoaderFactory.getClassLoader(type) : type.getClassLoader())
-          .getLoaded());
+          .make();
+      final ClassLoadingStrategy<ClassLoader> classLoadingStrategy = chooseClassLoadingStrategy(type);
+      if (classLoadingStrategy != null) {
+        return OBJENESIS.newInstance(unloaded
+            .load(useOSGiClassLoaderBridging ? BridgeClassLoaderFactory.getClassLoader(type) : type.getClassLoader(), classLoadingStrategy)
+            .getLoaded());
+      } else {
+        return OBJENESIS.newInstance(unloaded
+            .load(useOSGiClassLoaderBridging ? BridgeClassLoaderFactory.getClassLoader(type) : type.getClassLoader())
+            .getLoaded());
+      }
     } catch (Throwable t) {
       throw errors.errorInstantiatingProxy(type, t).toException();
     }
   }
-}
\ No newline at end of file
+
+  private static Method privateLookupIn;
+  private static Object lookup;
```

Comment:
can be `static final` and put them to the top of class.

Reply:
Actually, those two cannot be final, so I only moved those 2 up to the top.

Reply:
Please refer to [ClassLoadingStrategyFactory.java](https://github.com/joel-costigliola/assertj-core/blob/0b623f70bbe9de4d4785b7b3317528595d898c78/src/main/java/org/assertj/core/api/ClassLoadingStrategyFactory.java)

Reply:
Sure. 👌 👍